large_image/tilesource/base.py

```diff
@@ -182,16 +182,24 @@ def _setStyle(self, style):
         self._jsonstyle = style
         if style:
             if isinstance(style, dict):
-                self.style = style
+                self._style = JSONDict(style)
                 self._jsonstyle = json.dumps(style, sort_keys=True, separators=(',', ':'))
             else:
                 try:
-                    self.style = json.loads(style)
-                    if not isinstance(self.style, dict):
+                    self._style = JSONDict(json.loads(style))
```

Comment:
This will report an odd exception rather than a type error.  It'd produce more expected results to do
```
style = json.loads(style)
if not isinstance(style, dict):
    raise TypeError
self._style = JSONDict(style)
```